### ISSUE TITLE
s3 serve: incorrect format of Last-Modified header

### ISSUE THREAD

**Issue by Butanediol** [rclone/rclone#7762]
#### The associated forum post URL from `https://forum.rclone.org`

None

#### What is the problem you are having with rclone?

I'm building an app that uses S3 API and I'm testing it with `rclone serve s3`. But my app failed to download files through it. It crashed on decoding the `Last-Modified` header.

I'm using a [relatively mature library](https://github.com/soto-project/soto) to make S3 calls. (Relative to rclone, since the `serve s3` is still [being marked "experimental"](https://rclone.org/commands/rclone_serve_s3/)) . :)

After some investigation, I found that rclone returns the `Last-Modified` header in an unusual way. Instead of the more common `Thu, 01 Dec 1994 16:00:00 GMT` format, it returns [`Mon, 2 Jan 2006 15:04:05.999999999 GMT` (Link of code)](https://github.com/rclone/rclone/blob/51582e36e86d6be656cab7cc6b987fc8fb9688ed/cmd/serve/s3/backend.go#L21).

I then searched some AWS documentation but only found the following statement [link](https://docs.aws.amazon.com/sdkfornet1/latest/apidocs/html/P_Amazon_S3_Model_S3Object_LastModified.htm):

> The date retrieved from S3 is in ISO8601 format. **A GMT formatted date is passed back to the user.**

So I assume it is the same as the [HTTP Last-Modified header (Mozilla Doc)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Last-Modified)?

> ```http
> Last-Modified: <day-name>, <day> <month> <year> <hour>:<minute>:<second> GMT
> ```

I don't have an AWS S3 bucket to test this, but I search some response examples from the Internet. They are all in the more common way.

#### What is your rclone version (output from `rclone version`)

```
rclone v1.66.0
- os/version: darwin 14.4 (64 bit)
- os/kernel: 23.4.0 (arm64)
- os/type: darwin
- os/arch: arm64 (ARMv8 compatible)
- go/version: go1.22.1
- go/linking: dynamic
- go/tags: none
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

MacOS arm64 and x86_64 Linux Docker

#### Which cloud storage system are you using? (e.g. Google Drive)

None

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

`rclone server s3 .`

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

```
2024/04/14 17:52:52 DEBUG : rclone: Version "v1.66.0" starting with parameters ["rclone" "serve" "s3" "." "-vv"]
2024/04/14 17:52:52 DEBUG : Creating backend with remote "."
2024/04/14 17:52:52 DEBUG : Using config file from "/Users/<myusername>/.config/rclone/rclone.conf"
2024/04/14 17:52:52 DEBUG : fs cache: renaming cache item "." to be canonical "/Volumes/ACASIS"
2024/04/14 17:52:52 INFO  : Local file system at /Volumes/<MyDriveName>: poll-interval is not supported by this remote
2024/04/14 17:52:52 NOTICE: serve s3: No auth provided so allowing anonymous access
2024/04/14 17:52:52 NOTICE: Local file system at /Volumes/<MyDriveName>: Starting s3 server on [http://127.0.0.1:8080/]
2024/04/14 17:52:54 DEBUG : serve s3: LIST BUCKET
2024/04/14 17:52:54 DEBUG : serve s3: bucketname: pre prefix: prefix:"" page: {Marker: HasMarker:false MaxKeys:1000}
2024/04/14 17:52:54 DEBUG : serve s3: HEAD OBJECT pre filename.mp4
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
I think you are correct - this is out of spec.

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.67 and v1.66.1

Thank you @Butanediol
